src/modules/authorization/sagas.ts

```diff
@@ -155,6 +183,91 @@ export function createAuthorizationSaga() {
     }
   }
 
+  function* handleAuthorizationFlowRequest(
+    action: AuthorizationFlowRequestAction
+  ) {
+    const { authorizationAction, authorization, allowance } = action.payload
+    const isRevoke = authorizationAction === AuthorizationAction.REVOKE
+    const tokenRequest = isRevoke
+      ? revokeTokenRequest(authorization)
+      : grantTokenRequest(authorization)
+    const TOKEN_SUCCESS = isRevoke ? REVOKE_TOKEN_SUCCESS : GRANT_TOKEN_SUCCESS
+    const TOKEN_FAILURE = isRevoke ? REVOKE_TOKEN_FAILURE : GRANT_TOKEN_FAILURE
+
+    yield put(tokenRequest)
+
+    try {
+      const {
+        success,
+        failure
+      }: {
+        success: RevokeTokenSuccessAction | GrantTokenSuccessAction | undefined
+        failure: RevokeTokenFailureAction | RevokeTokenFailureAction | undefined
+      } = yield race({
+        success: take(TOKEN_SUCCESS),
+        failure: take(TOKEN_FAILURE)
+      })
+      if (failure) {
+        throw new Error(failure.payload.error)
+      }
+      const txHash = getTransactionFromAction(success!).hash
+
+      yield call(waitForTx, txHash)
+
+      yield put(fetchAuthorizationsRequest([authorization]))
+
+      const {
+        fetchFailure
+      }: {
+        fetchSuccess: FetchAuthorizationsSuccessAction | undefined
+        fetchFailure: FetchAuthorizationsFailureAction | undefined
+      } = yield race({
+        fetchSuccess: take(FETCH_AUTHORIZATIONS_SUCCESS),
+        fetchFailure: take(FETCH_AUTHORIZATIONS_FAILURE)
+      })
+
+      if (fetchFailure) {
+        throw new Error(
+          fetchFailure.payload.error ||
+            AuthorizationError.FETCH_AUTHORIZATIONS_FAILURE
+        )
+      }
+      const authorizations: Authorization[] = yield select(getData)
+
+      if (
+        authorizationAction === AuthorizationAction.REVOKE &&
+        hasAuthorization(authorizations, authorization)
+      ) {
+        throw new Error(AuthorizationError.REVOKE_FAILED)
+      }
+
+      if (authorizationAction === AuthorizationAction.GRANT) {
+        if (
+          authorization.type === AuthorizationType.ALLOWANCE &&
+          allowance &&
+          !hasAuthorizationAndEnoughAllowance(
+            authorizations,
+            authorization,
+            allowance
+          )
+        ) {
+          throw new Error(AuthorizationError.INSUFFICIENT_ALLOWANCE)
+        }
+
+        if (
+          authorization.type === AuthorizationType.APPROVAL &&
+          !hasAuthorization(authorizations, authorization)
+        ) {
+          throw new Error(AuthorizationError.GRANT_FAILED)
+        }
+      }
```

Comment:
Would you mind adding tests for these parts of the code? I might be wrong, but I don't think they're covered.